src/Workspaces/CSharp/Portable/Recommendations/CSharpRecommendationServiceRunner.cs

```diff
@@ -527,25 +527,52 @@ private RecommendedSymbols GetSymbolsOffOfConditionalReceiver(ExpressionSyntax o
             var useBaseReferenceAccessibility = symbol is IParameterSymbol { IsThis: true } p && !p.Type.Equals(containerType);
             var symbols = GetMemberSymbols(containerSymbol, position: originalExpression.SpanStart, excludeInstance, useBaseReferenceAccessibility, unwrapNullable, isForDereference);
 
-            // If we're showing instance members, don't include nested types
-            var namedSymbols = excludeStatic
-                ? symbols.WhereAsArray(s => !(s.IsStatic || s is ITypeSymbol))
-                : symbols;
-
-            //If container type is "ref struct" then we should exclude methods from object and ValueType that are not overriden
-            //if recomendations are requested not in nameof context,
-            //because calling them produces a compiler error due to unallowed boxing. See https://github.com/dotnet/roslyn/issues/35178
-            if (excludeBaseMethodsForRefStructs && containerType is not null && containerType.IsRefLikeType)
-            {
-                namedSymbols = namedSymbols.RemoveAll(s => s.ContainingType.SpecialType is SpecialType.System_Object or SpecialType.System_ValueType);
-            }
```

Comment:
instead of multiple passes that realize intermediary arrays each time, we do this as a single pass.